### ISSUE TITLE
Files: Spaces in paths not supported for all endpoints

### ISSUE THREAD

**Issue by scblack321** [tapis-project/tapis-files#79]
Shayan Khan at TACC reported problems with encoding spaces in paths for some of the endpoints.

> Shayan Khan
>   [7 minutes ago](https://tacc-team.slack.com/archives/C0307MGPCP2/p1677864257684219?thread_ts=1677864240.525339&cid=C0307MGPCP2)
> Currently before we make a request such as copy, move etc. we encode the path value to deal with special characters (e.g space in the path value gets encoded to %20). This has been working fine for the most part (and has been working fine in v2) with the operations except for a couple places where I am noticing weird behavior:
> 1- When I create a new directory using the mkdir command, the encoding persists in the final folder name (e.g New Folder shows up as New%20Folder).
> 2- Another bug I noticed was when using the createTransferTask() command with a file that has spaces, it errors out because it can't find the source (e.g if the filename is test file.txt it fails because it's looking for test%20file.txt because the source_path we sent in the request was encoded)  .
> I wanted to confirm a couple things:
> a- Are the issues I am seeing bugs on the tapis side or is this expected behavior? (Is it because the couple of commands I am seeing this issue on are POST requests?)
> b- Do we even need to deal with encoding these path values on our end in general? (We were doing this in v2 and copied the code when implementing the v3 commands)

**Comment by dvernon-tacc** [comment]
I'm still looking at this to see if maybe there are other issue, however I did test making a directory.  I used this:
**HTTPie**
 `http POST "https://dev.staging.tapis.io/v3/files/ops/tapisv3-storage-dev" path="dvernon/Directory With Spaces" X-Tapis-Token:$JWT`

**curl**
`curl -X POST "https://dev.staging.tapis.io/v3/files/ops/tapisv3-storage-dev" -d '{"path":"dvernon/Directory With Spaces"}' -H "X-Tapis-Token:$JWT" -H "Content-Type: application/json"`

**python**
`tapis.files.mkdir(systemId="tapisv3-storage-dev", path="dvernon/Directory With Spaces")`

This made a directory called "Directory With Spaces" without any problems.  Could it be possible that the path was supplied with url encoding in the json document payload?

I will do more testing - it's entirely possible that other endpoints don't handle the spaces.

It sounds like we might be inconsistent about if we are expecting url encoding.  I will also check that we are consistent.  I think the body shouldn't need it, but of course you will have to have it in the url.
